### ISSUE TITLE
Can't find how to find out my username given my email adress

### ISSUE THREAD

**Issue by davidascher** [mozilla/id.webmaker.org#281]
The 'create an account' complains if I give my email address as david.ascher@gmail.com, but I can't get the system to tell me my username given that email address.

**Comment by secretrobotron** [comment]
Relates to https://github.com/mozilla/id.webmaker.org/issues/283#issuecomment-96814784

**Comment by cassiemc** [comment]
Can we allow people to log in with either their username or email? @secretrobotron I know we talked about this yesterday, and thinking again this morning it feels like the right UX to me. This way we don't even really need a flow for 'lost username' because they would need their email anyway. Perhaps we could adjust and add some copy to the 'forgot password' flow. /cc @jbuck

**Comment by cassiemc** [comment]
@alicoding I can't remember why we changed some of the wording in the flow. I am mocking up a ux flow to help clarify, just a quick design review with jess first.

**Comment by secretrobotron** [comment]
Yep, that sounds right to me for short term. Yay minimalism! +1 to adding copy to forgot password flow if needed.

@thisandagain originally wanted to implement multiple accounts per email, but we don't/can't support that atm (see #193).

**Comment by secretrobotron** [comment]
@simonwex can you spare a dev to help here? @alicoding has most experience with this code atm. I'm happy to help & review.

**Comment by cassiemc** [comment]
@alicoding @secretrobotron This feels like the right flow and is mostly what we have so far, though we need to tweak language and add headlines / more text to the pages to clarify what's happening.  

One more thing to consider is those users who haven't yet set a password (not sure what % of our users that actually is)
![forgot-username-02](https://cloud.githubusercontent.com/assets/1186326/7394091/792303d6-ee5f-11e4-87c2-f52b4fa40826.png)

**Comment by secretrobotron** [comment]
Before @cassiemc's internet imploded, we chatted about an alternate flow, which more closely matches what we currently have. Will catch up with her more when she re-appears.

**Comment by cassiemc** [comment]
Sorry about that @secretrobotron! Here's the revised flow as we were discussing it, now with two links – one for 'forgot your username', and another for 'reset password'. I still think the first flow was more intuitive but this more closely matches what we've built.

![forgot-username-v3-03](https://cloud.githubusercontent.com/assets/1186326/7395835/ac1d50d8-ee6a-11e4-930c-b84d82144fd2.png)

**Comment by secretrobotron** [comment]
Rad. Thanks @cassiemc . I'll chat with @alicoding (provided @simonwex has no argument) to check in on implementing.

**Comment by cassiemc** [comment]
No problem. I have no qualms if we need to make adjustments, just ping me.

**Comment by simonwex** [comment]
@secretrobotron, @alicoding needs to get on with some scheduled work with @edrushka. I'll try to find someone else to help.

**Comment by secretrobotron** [comment]
(｀_´)ゞ

**Comment by secretrobotron** [comment]
Needs:

#### id.webmaker.org
* New page and react routing to match design
* Route to accept "forgot username" requests

#### webmaker-mailroom
* New template for "forgot username" (variant of reset password)

#### sawmill
* Update webmaker-mail package
* Set up logic to use "forgot username" template
  * see https://github.com/mozilla/sawmill/blob/master/worker/reset_request.js#L11
  * see https://github.com/mozilla/sawmill/blob/master/worker/login_request.js#L21 for example of how this was done for login emails

**Comment by kumarrishav** [comment]
https://github.com/mozilla/webmaker-android/issues/2320
